website/pages/docs/k8s/connect/ingress-gateways.mdx

```diff
@@ -5,4 +5,174 @@ sidebar_title: Ingress Gateways
+[https://releases.hashicorp.com/consul/1.8.0/](https://releases.hashicorp.com/consul/1.8.0/)
+
+-> If TLS and ACLs are enabled, you must set the following environment variables.
+
+```shell-session
+$ export CONSUL_HTTP_ADDR=https://localhost:8501
+$ export CONSUL_HTTP_TOKEN=$(kubectl get secret <consul-release>-bootstrap-acl-token -o jsonpath={.data.token} | base64 -D)
+$ export CONSUL_HTTP_SSL_VERIFY=false
+```
+
+### Submitting an Ingress Gateway configuration entry
+
+Now that you have access to the Consul cli you are able to submit an ingress gateway configuration entry.
+Here is an example service definition.
+
+```shell-session
+$  cat > ingress-gateway.hcl <<EOF
+Kind = "ingress-gateway"
+Name = "ingress-gateway"
+
+Listeners = [
+ {
+   Port = 8080
+   Protocol = "http"
+   Services = [
+     {
+       Name = "static-server"
+     }
+   ]
+ }
+]
+EOF
+```
+
+Submit the ingress gateway service definition with the Consul cli using this command.
+
+```shell-session
+$ consul config write ingress-gateway.hcl
+```
+
+-> You can confirm service registration by looking at the consul-ui : http://localhost:8500/ui/dc1/services/ingress-gateway/
+
+## Defining an Intention
+
+If ACLs are enabled, you can define an [intention](https://www.consul.io/docs/commands/intention) to allow the ingress gateway to access the upstream services defined in the config entries.
+For detailed instructions on how to configure zero-trust networking with intentions please refer to this [guide](https://learn.hashicorp.com/consul/gs-consul-service-mesh/network-security-with-consul-service-mesh).
+
+-> If you are using ACLs, you must create an intention to enable inter-service communication.
+
+## Deploying your application to Kubernetes
+Now you will deploy a sample application which echos “hello world”
+
+```shell-session
+$  cat > static-server.yaml <<EOF
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: static-server
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: static-server
+  annotations:
+    'consul.hashicorp.com/connect-inject': 'true'
+spec:
+  containers:
+    # This name will be the service name in Consul.
+    - name: static-server
+      image: hashicorp/http-echo:latest
+      args:
+        - -text="hello world"
+        - -listen=:8080
+      ports:
+        - containerPort: 8080
+          name: http
+    # If ACLs are enabled, the serviceAccountName must match the Consul service name.
+  serviceAccountName: static-server
+EOF
+```
+
+```shell-session
+kubectl apply -f static-server.yaml
+```
+
+## Connecting to your application
+
+You can validate the connectivity of the application in the consul-ui by navigating to
+`http://localhost:8500/ui/dc1/services/static-server/instances`
+
+You can also validate the connectivity of the application using a curl command
+```shell-session
+EXTERNAL_IP=kubectl get services | grep ingress-gateway | awk ‘{print $4}’
+curl -H "Host: static-server.ingress.consul" $EXTERNAL_IP:8080
+```
```

Comment:
Can you add a clean up step, this leaves them with an unauthenticated load balancer routing plaintext into their cluster